pkg/snmp/traps/listener_test.go

```diff
@@ -121,6 +140,42 @@ func TestServerV3BadCredentials(t *testing.T) {
 	assertNoPacketReceived(t, trapListener)
 }
 
+func TestListenerTrapsReceivedTelemetry(t *testing.T) {
+	mockSender := mocksender.NewMockSender("snmp-traps-telemetry")
+	mockSender.SetupAcceptAll()
+
+	config := Config{Port: serverPort, CommunityStrings: []string{"public"}, Namespace: "totoro"}
+	Configure(t, config)
+
+	packetOutChan := make(PacketsChannel)
+	trapListener, err := startSNMPTrapListener(config, mockSender, packetOutChan)
+	require.NoError(t, err)
+	defer trapListener.Stop()
+
+	sendTestV1GenericTrap(t, config, "public")
+	packet := receivePacket(t, trapListener) // Wait for packet
+	require.NotNil(t, packet)
+	mockSender.AssertMetric(t, "Count", "datadog.snmp_traps.received", 1, "", []string{"snmp_device:127.0.0.1", "namespace:totoro"})
+}
+
+func TestListenerUnknownCommunityStringTelemetry(t *testing.T) {
+	mockSender := mocksender.NewMockSender("snmp-traps-telemetry")
+	mockSender.SetupAcceptAll()
+
+	config := Config{Port: serverPort, CommunityStrings: []string{"public"}, Namespace: "totoro"}
+	Configure(t, config)
+
+	packetOutChan := make(PacketsChannel)
+	trapListener, err := startSNMPTrapListener(config, mockSender, packetOutChan)
+	require.NoError(t, err)
+	defer trapListener.Stop()
+
+	sendTestV1GenericTrap(t, config, "notpublic") // Send trap with invalid community string
+	time.Sleep(3 * time.Second)
```

Comment:
For reviewers: I'm a bit afraid of introducing flakiness with this, WDYT?

Reply:
How long does it usually take? You could try looping until it either succeeds or times out after X seconds. This way if it usually takes 2 seconds, you can timeout after 5 or 10 and it should help with the flakiness while keeping the tests "quick" in most cases.

Reply:
In NetFlow I had similar case. We are are waiting for the flows to be received and as soon as it's received, we can continue the test. Example:

https://github.com/DataDog/datadog-agent/blob/c782dbc3413d5a708d6a79cb345183c33d9718ad/pkg/netflow/server_integration_test.go#L73

Wait function: https://github.com/DataDog/datadog-agent/blob/77a596931aa9faa67ae398be252bfcd4519e7338/pkg/netflow/flowaggregator/testutil.go#L13-L31

Reply:
From what I've seen on average there is 2ms between when the packet is sent and processed, so 3 second is indeed too much.
WDYT of this wait function ? Since packet has an invalid community string, it is not sent to the packet channel so I cannot wait on this. I can however wait until the expvar `trapsPacketsAuthErrors` has been incremented :
```go
func WaitForInvalidPacketToBeReceived(t *testing.T, timeoutDuration time.Duration) {
	timeout := time.After(timeoutDuration)
	ticker := time.NewTicker(100 * time.Millisecond)
	defer ticker.Stop()
	for {
		select {
		// Got a timeout! fail with a timeout error
		case <-timeout:
			t.Error("timeout error waiting for packet")
		// Got a tick, we should check if packet has been received
		case <-ticker.C:
			if trapsPacketsAuthErrors.Value() > 0 {
				return
			}
		}
	}
}
```

Reply:
`trapsPacketsAuthErrors` might be too specific and would only work for error cases.

Maybe you can add a atomic counter for received packets? Example:

https://github.com/DataDog/datadog-agent/blob/c782dbc3413d5a708d6a79cb345183c33d9718ad/pkg/netflow/flowaggregator/aggregator.go#L45

Reply:
Yes, this wait function should wait for invalid packets to be received. For valid packets, we already have this :
https://github.com/DataDog/datadog-agent/blob/8a8ef53fe4e38a85d05876580618425666034113/pkg/snmp/traps/listener_test.go#L179-L188
But it doesn't work for invalid packets because they are not transmitted to the `listener.packets` channel